### ISSUE TITLE
Router-state plugin

### ISSUE THREAD

**Issue by filipjnc** [datorama/akita#23]
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report
[ ] Performance issue
[x] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## What is the motivation / use case for changing the behavior?
I am considering Akita for a fairly complex app and I seem to miss just a single puzzle piece in migrating from ngrx/ngxs - the router state. This plugin would address two important use cases in my view:
1. Memoized selectors based on route params and queryParams (e.g. active entity based on id param)
2. Time travel with Redux DevTools & overall transparency while debugging (Router_Navigate, Router_Cancel, etc.)

Would it fit well in Akita's infrastructure?

**Comment by NetanelBasal** [comment]
It seems like a good idea. Will need to investigate it.

**Comment by filipjnc** [comment]
@NetanelBasal A rough idea how I synchronized the router with the router-state:
https://stackblitz.com/edit/akita-todos-app-ge8rcp?file=src%2Fapp%2Frouter%2Frouter.service.ts

It's just a quick dirty implementation and there are many things open:
1. Navigation Cancel and Navigation Error: only the methods with custom actions are there, I got to figure out how to update the state in each case
2. The ability to provide a custom router serializer: the plugin would need to provide some interfaces/abstract classes to implement
3. Transform to plugin 😃
